File: solutions/lua/isbn-verifier/1/isbn-verifier.lua
return {
  valid = function(isbn)
    local tmp = string.gsub(isbn, "%W+", "")
    if #tmp ~= 10 then
      return false
    end

    local tmp02 = {}
    tmp:gsub(".", function(c)
      table.insert(tmp02, c)
    end)

    local test = tmp02[10]:find("[0-9X]+")
    if test == nil then
      return false
    end
    if tmp02[10] == "X" then
      tmp02[10] = ":"
    end

    local ii = 1
    local check = 0
    for i = 10, 1, -1 do
      check = check + ((tmp02[ii]:byte() - ("0"):byte()) * i)
      ii = ii + 1
    end

    local result = check % 11
    if result == 0 then
      return true
    end
    return false
  end
}

File: solutions/go/isbn-verifier/1/isbn_verifier.go
package isbn

import (
	"strings"
)

func IsValidISBN(isbn string) bool {
	isbn = strings.ReplaceAll(isbn, "-", "")
	if len(isbn) == 0 || len(isbn) != 10 {
		return false
	}
	tmp := strings.IndexRune(isbn, 'X')
	if tmp < 9 && tmp > 0 {
		return false
	}
	isbn = strings.Replace(isbn, "X", string('9'+1), 1)
	result := 0
	for i, v := range isbn {
		result += int(v-'0') * (10 - i)
	}
	return result%11 == 0
}
